test/boost/statement_restrictions_test.cc

```diff
@@ -359,3 +359,136 @@ SEASTAR_TEST_CASE(slice_single_column_mixed_order) {
                     singular({I(1), I(1)}), singular({I(1), I(2)})}));
     });
 }
+
+// Currently expression doesn't have operator==().
+// Implementing it is ugly, because there are shared pointers and the term base class.
+// For testing purposes checking stringified expressions is enough.
+static bool expression_eq(const expr::expression& e1, const expr::expression& e2) {
+    return to_string(e1) == to_string(e2);
+}
+
+static void assert_expr_vec_eq(const std::vector<expr::expression>& v1, const std::vector<expr::expression>& v2) {
+    if (std::equal(v1.begin(), v1.end(), v2.begin(), v2.end(), expression_eq)) {
+        return;
+    }
+
+    BOOST_FAIL(fmt::format("Expression vectors not equal! [{}] != [{}]", fmt::join(v1, ", "), fmt::join(v2, ", ")));
```

Comment:
This isn't enough to unambiguously show me which test failed (when one does fail).  It needs source_location of the caller, like require_rows().

Reply:
I added the location, now a sample error message looks like this:
```
test/boost/statement_restrictions_test.cc(382): fatal error: in "expression_extract_column_restrictions": Location: test/boost/statement_restrictions_test.cc:483, Expression vectors not equal! [(pk2) = 00000000, ((pk1,pk2)) < (00000000, 00000000)] != [(pk2) = 00000000]
```

Reply:
Try it in an IDE, see if it parses that error message and lets you jump straight to the source in one click.  I did that for require_rows().

Reply:
VSCode allows me to jump straight to source by doing `Ctrl+Click` on `test/boost/statement_restrictions_test.cc:483`

Reply:
Great, ship it! ;)